pkg/client/cluster.go

```diff
@@ -18,15 +17,63 @@ func (c *client) Ping(vsn string) error {
 	return err
 }
 
-func (c *client) RegisterRuntimeServices(svcs map[string]string, serviceId *string) error {
+func initLayouts(layouts *console.OperationalLayoutAttributes) *console.OperationalLayoutAttributes {
+	if layouts == nil {
+		return &console.OperationalLayoutAttributes{
+			Namespaces: &console.ClusterNamespacesAttributes{},
+		}
+	}
+	return layouts
+}
+
+func appendUniqueExternalDNSNamespace(slice []*string, newValue *string) []*string {
+	if slice == nil {
+		slice = make([]*string, 0)
+	}
+	for _, val := range slice {
+		if val == newValue {
+			return slice
+		}
+	}
+	// Append the unique value
+	return append(slice, newValue)
+}
+
+func (c *client) RegisterRuntimeServices(svcs map[string]*NamespaceVersion, serviceId *string) error {
 	inputs := make([]*console.RuntimeServiceAttributes, 0)
-	for name, version := range svcs {
+	var layouts *console.OperationalLayoutAttributes
+	for name, nv := range svcs {
 		inputs = append(inputs, &console.RuntimeServiceAttributes{
 			Name:    name,
-			Version: version,
+			Version: nv.Version,
 		})
+		switch name {
+		case "cert-manager":
```

Comment:
can we move these strings to constants (mostly for code cleanliness)

Reply:
done